benedict/dicts/keylist/keylist_dict.py

```diff
@@ -43,7 +43,7 @@ def __getitem__(self, key):
 
     def _getitem_by_keys(self, keys):
         parent, key, _ = keylist_util.get_item(self, keys)
-        if type_util.is_list_of_dicts(parent):
+        if type_util.is_list_of_dicts(parent) and any(type_util.is_wildcard(_key) for _key in keys):
```

Comment:
I'm pretty sure that this check `any(type_util.is_wildcard(_key) for _key in keys)` appears multiple times, and since there is also some logic involved in the check I think it's time to add a new `type_util` method like:

```python
def any_wildcard_in_list(val):
    return any(is_wildcard(item) for item in val)
```